src/io/flutter/editor/outline/OpenEditorOutlineService.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2019 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.editor.outline;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.ex.EditorEx;
+import com.intellij.openapi.fileEditor.*;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.io.FileUtil;
+import com.intellij.openapi.vfs.VirtualFile;
+import io.flutter.FlutterUtils;
+import io.flutter.dart.FlutterDartAnalysisServer;
+import io.flutter.dart.FlutterOutlineListener;
+import org.dartlang.analysis.server.protocol.FlutterOutline;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+import java.util.*;
+
+/**
+ * Service that watches the Flutter Outlines for all open Dart files.
+ *
+ * <p>
+ * This service works by listening to the {@link Project}'s MessageBus for {@link FileEditor}s that are
+ * added or removed. Using the set of currently active editor windows ({@link EditorEx}s), this service
+ * then subscribes to the {@link FlutterDartAnalysisServer} for updates to the {@link FlutterOutline} of each file.
+ */
+public class OpenEditorOutlineService implements Disposable {
```

Comment:
The word open is a bit confusing to me as it made me thing this service opens editor outlines not that it is showing outlines for the visible editors. How about
ActiveEditorOutlineService or SelectedEditorOutlineService?